### PR TITLE
feat(Scalar.AspNetCore): support multiple scalar endpoints

### LINKED ISSUE #3901: Cannot map multiple endpoints
### What happens?

Using the ASP.NET Core package I generate two OpenAPI documents as my API has two clients with different authentication schemes and wanted to map both to separate endpoints. 

```
       app.MapScalarApiReference(options =>
            {
                options.WithEndpointPrefix("/scalar/admin/{documentName}");
                options.WithOpenApiRoutePattern("/openapi/admin.json");
                options.WithTitle("Admin");
                options.WithOAuth2Authentication(authOptions =>
                {
                    authOptions.ClientId = app.Configuration["Admin:ClientId"];
                    authOptions.Scopes = [app.Configuration["Admin:Scope"]!];
                });
            })
            .AllowAnonymous();

        app.MapScalarApiReference(options =>
            {
                options.WithEndpointPrefix("/scalar/partner/{documentName}");
                options.WithOpenApiRoutePattern("/openapi/partner.json");
                options.WithTitle("Partner");
                options.WithOAuth2Authentication(options =>
                {
                    options.ClientId = app.Configuration["Partner:ClientId"];
                    options.Scopes = [app.Configuration["Partner:Scope"]!];
                });
            })
            .AllowAnonymous();
```

However, the MapScalarApiReference using GetService<IOptions<ScalarOptions>>, which will return the same reference for both calls by default, so both endpoints ends up using the same options object with the values from the second configuration of the options.

### What did you expect to happen?

I expected be able to configure the two endpoints independently.

I worked around this by forcing the options to be null:

builder.Services.AddTransient<IOptions<ScalarOptions>>(sp => null!);

But I think perhaps it would be nice with a more obvious way to do this. 

### OpenAPI Document

_No response_

### PR DESCRIPTION
**Pull request by xC0dex**
This PR introduces support for registering multiple `MapScalarApiReference` endpoints. To achieve this, I decided to clone the options before invoking them in the action. This approach ensures that the options can be invoked multiple times without side effects.

Closes #3901

### CODE REVIEW

File: packages/scalar.aspnetcore/src/Scalar.AspNetCore/Extensions/ScalarEndpointRouteBuilderExtensions.cs [hunk 1]
```diff
@@ -36,7 +36,8 @@ public static IEndpointConventionBuilder MapScalarApiReference(this IEndpointRou
     /// <param name="configureOptions">An action to configure the Scalar options.</param>
     public static IEndpointConventionBuilder MapScalarApiReference(this IEndpointRouteBuilder endpoints, Action<ScalarOptions> configureOptions)
     {
-        var options = endpoints.ServiceProvider.GetService<IOptions<ScalarOptions>>()?.Value ?? new ScalarOptions();
+        // Clone existing ScalarOptions or create a new instance
+        var options = endpoints.ServiceProvider.GetService<IOptions<ScalarOptions>>()?.Value is { } existingOptions ? existingOptions with { } : new ScalarOptions();
```
[the remark this answers is not in the document]
Comment (xC0dex) [reply packages/scalar.aspnetcore/src/Scalar.AspNetCore/Extensions/ScalarEndpointRouteBuilderExtensions.cs:40]: Hey @alfkonee, thanks for the review! I had the same thought about using named options for each OpenAPI document name. However, during the `MapScalarApiReference` call, we don't know the name of the OpenAPI document. The options are needed to correctly map the endpoint route based on the `EndpointPathPrefix`.

Once Scalar supports multiple API documents, we can revisit this package and refactor it to add/configure OpenAPI documents via the options. At that point, introducing named options may also make sense.

File: packages/scalar.aspnetcore/src/Scalar.AspNetCore/Options/ScalarOptions.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@ namespace Scalar.AspNetCore;
 /// Represents all available options for the Scalar API reference.
 /// Based on <a href="https://github.com/scalar/scalar/blob/main/documentation/configuration.md">Configuration</a>.
 /// </summary>
-public sealed class ScalarOptions
+public sealed record ScalarOptions
```
Comment (xC0dex) [inline packages/scalar.aspnetcore/src/Scalar.AspNetCore/Options/ScalarOptions.cs:7]: I'm not sooo happy with this. I'll rethink this solution.

**Comment by DaleMckeown** [comment]
@xC0dex Will this also enable Scalar to pull OpenApi specs from remote hosts?

Imagine you have 5 microservices on their own URLs, each exposing an OpenApi spec. Then you have one documentation site that wants to consume all of those OpenApi spec documents and show them in a single scalar UI.

This is possible in Swagger using the `swaggerOptions.SwaggerEndpoint` option, but I can't seem to find anything similar for Scalar.

**Review by hanspagel — CHANGES_REQUESTED** [review]
This is awesome! Can you add this to the README, too? 🤩

**Comment by xC0dex** [comment]
I think I'll close this PR and open a new one once I have a suitable solution for it.

**Pull request closed without merging**